core/src/main/scala/cats/data/RepresentableStore.scala

```diff
@@ -0,0 +1,52 @@
+package cats.data
+
+import cats.{Comonad, Functor, Representable}
+
+/**
+ * A generalisation of the Store comonad, for any `Representable` functor.
+ * `Store` is the dual of `State`
+ */
+final case class RepresentableStore[F[_], S, A](fa: F[A])(val index: S)(implicit R: Representable.Aux[F, S]) {
```

Comment:
I would rather not use the second parameter for index. I think scala does not use that for equality (only the first item is checked) and I doing think we want that. Also, I’d rather take R as an implicit parameter at the callsites that use it.